core/chains/evm/logpoller/log_poller.go

```diff
@@ -276,163 +269,177 @@ func (lp *logPoller) backfill(ctx context.Context, start, end int64) int64 {
 	return end + 1
 }
 
-func (lp *logPoller) maybeHandleReorg(ctx context.Context, currentBlockNumber, latestBlockNumber int64) (*types.Block, bool, int64, error) {
+// getCurrentBlockMaybeHandleReorg accepts a block number
+// and will return that block if its parent points to our last saved block.
+// If its parent does not point to our last saved block we know a reorg has occurred.
+// In that case return the LCA+1, i.e. our new current (unprocessed) block.
+func (lp *logPoller) getCurrentBlockMaybeHandleReorg(ctx context.Context, currentBlockNumber int64) (*types.Block, error) {
 	currentBlock, err1 := lp.ec.BlockByNumber(ctx, big.NewInt(currentBlockNumber))
 	if err1 != nil {
 		lp.lggr.Warnw("Unable to get currentBlock", "err", err1, "currentBlockNumber", currentBlockNumber)
-		return nil, false, currentBlockNumber, err1
+		return nil, err1
 	}
 	// Does this currentBlock point to the same parent that we have saved?
 	// If not, there was a reorg, so we need to rewind.
-	expectedParent, err1 := lp.orm.SelectBlockByNumber(currentBlockNumber - 1)
+	expectedParent, err1 := lp.orm.SelectBlockByNumber(currentBlockNumber-1, pg.WithParentCtx(ctx))
 	if err1 != nil && !errors.Is(err1, sql.ErrNoRows) {
 		// If err is not a no rows error, assume transient db issue and retry
 		lp.lggr.Warnw("Unable to read latestBlockNumber currentBlock saved", "err", err1, "currentBlockNumber", currentBlockNumber)
-		return nil, false, 0, errors.New("Unable to read latestBlockNumber currentBlock saved")
+		return nil, errors.New("Unable to read latestBlockNumber currentBlock saved")
 	}
-	// We will not have the previous currentBlock on initial poll or after a backfill.
-	havePreviousBlock := !errors.Is(err1, sql.ErrNoRows)
-	if havePreviousBlock && (currentBlock.ParentHash() != expectedParent.BlockHash) {
+	// We will not have the previous currentBlock on initial poll.
+	havePreviousBlock := err1 == nil
+	if !havePreviousBlock {
+		lp.lggr.Warnw("Do not have previous block, first poll ever on new chain")
```

Comment:
Nit: I think this can be safely downgraded to `Info`, since it's normal behavior on a first ever poll right?

Reply:
yeah fair, I should update it to say or first poll after backfill too